### PR TITLE
Fix circular dependency PetabImporter-PetabAmiciObjective; tidy up and extend docs

### PR DESCRIPTION
**Pull request by yannikschaelte**
Details:
* added possibility to recreate AmiciObjective via builder objects
* [not affecting API] Remove PetabAmiciObjective, since it gives no additional functionality anymore compared to AmiciObjective

### CODE REVIEW

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/petab/petab_importer.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5340943&bid=17486131)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/petab/petab_importer.py  5
- pypesto/objective/amici_util.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5340943&bid=17486131)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/objective/amici_util.py  5
- pypesto/petab/importer.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5340943&bid=17486131)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/petab/importer.py  5
- pypesto/objective/amici_util.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5340943&bid=17486131)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/objective/amici_util.py  5
- pypesto/petab/importer.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5340943&bid=17486131)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/petab/importer.py  5
- pypesto/objective/amici_util.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5340943&bid=17486131)

File: pypesto/objective/amici_objective.py [hunk 1]
```diff
@@ -223,6 +250,44 @@ def __deepcopy__(self, memodict: Dict = None) -> 'AmiciObjective':
...
+        self.__dict__.update(state)
+
+        # note: attributes not defined in the builder are lost
+        model = self.amici_object_builder.create_model()
+        solver = self.amici_object_builder.create_solver(model)
+        edatas = self.amici_object_builder.create_edatas(model)
+
+        amici.readSolverSettingsFromHDF5(
+            state['amici_solver_settings'], solver)
+        os.remove(state['amici_solver_settings'])
```
Comment (FFroehlich) [inline pypesto/objective/amici_objective.py:285]: still worried that this limits unpickling to once.

Comment (yannikschaelte) [reply pypesto/objective/amici_objective.py:285]: agreed, will be done in #278

**Pull request merged**